sql/src/main/java/io/crate/execution/engine/fetch/TransportFetchOperation.java

```diff
@@ -84,6 +84,13 @@ static RamAccounting ramAccountingForIncomingResponse(RamAccounting ramAccountin
             return RamAccounting.NO_ACCOUNTING;
         }
         // Each response may run in a different thread and thus should use its own ram accounting instance
-        return new BlockBasedRamAccounting(ramAccounting::addBytes, MAX_BLOCK_SIZE_IN_BYTES);
+        return new BlockBasedRamAccounting(
+            usedBytes -> {
+                synchronized (ramAccounting) {
```

Comment:
I think this is worth a comment something like "A non-thread-safe ram accounting instance may be used here, so we must ensure thread-safety."